lib/Models/RegionProviderList.js

```diff
@@ -0,0 +1,135 @@
+
+    /** 
+     * List of RegionProviders, once loaded from file. 
+     * @type {RegionProvider[]} 
+     */
+    this.regionProviders = [];
+
+    /**
+     * Has this list been initialized from file yet.
+     * @type {Boolean}
+     */
+    this._initialized = false;
+
+};
+
+/**
+ * Fetch JSON file and create ordered list of region providers from it. 
+ * @returns Promise.
+ */
+RegionProviderList.prototype.init = function(url) {
+    if (this._initialized) {
+        return when(this);
+    }
+    if (!defined(url)) {
+        url = this.terria.regionMappingDefinitionsUrl;
+    }
+
+    var that = this;
+    return loadJson(url)
+        .then(function(o) { 
+            return that.initFromObject(o);
+        });
+};
+
+/**
+ * Initialises from the already-retrieved contents of a JSON file.
+ */
+RegionProviderList.prototype.initFromObject = function(obj) {
+    if (this._initialized) {
+        throw new DeveloperError ("RegionProviderList already initialized.");
+    }
+    var that = this;
+    Object.keys(obj.regionWmsMap).forEach(function(r) {
+        that.regionProviders.push(new RegionProvider(r, obj.regionWmsMap[r], that));
+    });
+    this._initialized = true;
+    return this;
+};
+
+/**
+ * Has RegionProviderList finished loading config file?
+ * @returns {Boolean}
+ */
+RegionProviderList.prototype.ready = function() {
+    return this._initialized;
+};
+
+/**
+  *  Find what kind of region-mapped dataset this is by exhaustively looking for every alias in every column. 
+  *  @param varNames {String[]} Array of names of variables to choose amongst.
+  */
+RegionProviderList.prototype.chooseRegionProvider = function (varNames) {
+    if (!this._initialized) {
+        throw new DeveloperError('Region provider list not initialized.');
+    }
+    //try to figure out the region variable and type based on aliases
+    var i;
+    for (i = 0; i < this.regionProviders.length; i++) {
+        var idx = this.regionProviders[i].findRegionVariable(varNames);
+        if (idx !== -1) {
+            return { 
+                regionProvider: this.regionProviders[i], 
+                regionVariable: varNames[idx],
+                disambigVariable: this.regionProviders[i].findDisambigVariable(varNames)
+            };
+        }
+    }
+
+    return null;
+};
+
+/**
+    Get the region provider matching a regionType string (eg, STE)
+    @param {String} regionType The type string to look up.
+    @return {Object} Region Provider, or null.
+*/
+RegionProviderList.prototype.getRegionProvider = function(regionType) {
+    if (!this._initialized) {
+        throw new DeveloperError('Region mapping list not initialized.');
+    }
+    var r = this.regionProviders.filter(function(p) { return p.regionType.toLowerCase() === regionType.toLowerCase(); });
+    if (r.length > 1)
+        throw new DeveloperError ('More than one definition of region provider: ' + regionType);
+    if (r.length === 0)
+        return null;
```

Comment:
We usually use `undefined` instead of `null`.  JS conveniently has two "there's no object here" values.  Using one consistently (undefined) means we don't have to check for both all over the place.